mozdownload/scraper.py

```diff
@@ -331,17 +333,44 @@ class DailyScraper(Scraper):
     """Class to download a daily build from the Mozilla server."""
 
     def __init__(self, branch='mozilla-central', build_id=None, date=None,
-                 build_number=None, *args, **kwargs):
+                 build_number=None, revision=None, *args, **kwargs):
         """Create an instance of the daily scraper."""
         self.branch = branch
         self.build_id = build_id
         self.date = date
         self.build_number = build_number
+        self.revision = revision
 
         Scraper.__init__(self, *args, **kwargs)
 
     def get_build_info(self):
         """Define additional build information."""
+        # Retrieve build by revision
+        if self.revision:
+            th = treeherder.Treeherder(
+                APPLICATIONS_TO_FTP_DIRECTORY.get(self.application, self.application),
+                self.branch,
+                self.platform)
+            builds = th.query_builds_by_revision(
+                self.revision,
+                job_type_name='L10n Nightly' if self.locale_build else 'Nightly')
+
+            if not builds:
+                raise errors.NotFoundError('No builds have been found for revision', self.revision)
+
+            # Extract the build folders which are prefixed with the buildid
+            self.builds = [build.rsplit('/', 2)[1] for build in builds]
+            self.show_matching_builds(self.builds)
+
+            # There is only a single build per revision and platform
+            self.build_index = 0
+            self.logger.info('Selected build: %s' % self.builds[self.build_index])
+
+            self.date = datetime.strptime(self.builds[self.build_index][:19],
```

Comment:
Suggestion, feel free to ignore: it would be helpful to note your assumptions about the data you're working with, like `build`, so it's easier to debug future issues with expressions like `build.rsplit('/', 2)[1]` or `self.builds[self.build_index][:19]`.